code/modules/assembly/infrared.dm

```diff
@@ -10,101 +10,132 @@
 
 	wires = WIRE_PULSE
 
-	secured = 0
+	secured = TRUE
 
-	var/on = 0
-	var/visible = 0
+	var/on = FALSE
+	var/visible = TRUE
 	var/obj/effect/beam/infrared/beam = null
 
 	accessible_values = list(
 		"Visible" = "visible;number",\
 		"On" = "on;number"
 	)
 
-/obj/item/device/assembly/infra/New()
-	qdel(src) // Why is this even
+/obj/item/device/assembly/infra/New(var/turf/loc)
+	..()
+	update_icon()
 
-///obj/item/device/assembly/infra/describe()
-//	return "The infrared trigger is [on?"on":"off"]."
+/obj/item/device/assembly/infra/Destroy(var/turf/loc)
+	processing_objects.Remove(src)
+	if (beam)
+		returnToPool(beam)
+		beam = null
+	..()
+
+/obj/item/device/assembly/infra/examine(var/mob/user)
+	..()
+	to_chat(user, "<span class='notice'>The trigger is [on?"on":"off"].</span>")
+	to_chat(user, "<span class='notice'>The lens is facing [dir2text(dir)].</span>")
 
 /obj/item/device/assembly/infra/activate()
 	if(!..())
 		return 0//Cooldown check
 	on = !on
+	var/turf/T = get_turf(src)
+	if (T)
+		playsound(T,'sound/misc/click.ogg',30,0,-5)
+
+	if (on)
+		processing_objects.Add(src)
+		playsound(T,'sound/weapons/egun_toggle_laser.ogg',70,0,-5)
+		process()
+	else
+		if (beam)
+			returnToPool(beam)
+			beam = null
+		processing_objects.Remove(src)
+		playsound(T,'sound/weapons/egun_toggle_taser.ogg',70,0,-5)
 	update_icon()
 	return 1
 
-
 /obj/item/device/assembly/infra/toggle_secure()
 	secured = !secured
-	if(secured)
-		processing_objects.Add(src)
-	else
+	if(!secured)
 		on = 0
 		if(beam)
 			returnToPool(beam)
+			beam = null
 		processing_objects.Remove(src)
 	update_icon()
 	return secured
 
-
 /obj/item/device/assembly/infra/update_icon()
 	overlays.len = 0
```

Comment:
```suggestion
	overlays.Cut()
```

Reply:
can I not do that? I've been conditioned by @d3athrow to use len = 0
https://github.com/vgstation-coders/vgstation13/commit/72fde47f91f7d4fed3a34f0175a0ff8fdab0b043#diff-f8a195145812f9e4ace5ed0ff4643950

Reply:
Sure